src/wallet/wallet2.cpp

```diff
@@ -12994,6 +12996,26 @@ std::unordered_map<std::string, wallet2::ons_detail> wallet2::get_ons_cache()
   return ons_records_cache;
 }
 
+void wallet2::refresh_batching_cache()
+{
+  rpc::GET_ACCRUED_BATCHED_EARNINGS::request req{};
+  rpc::GET_ACCRUED_BATCHED_EARNINGS::response daemon_resp{};
+  bool r = invoke_http<rpc::GET_ACCRUED_BATCHED_EARNINGS>(req, daemon_resp);
+  if (r && daemon_resp.status == rpc::STATUS_OK)
+  {
+    batching_records_cache.clear();
+    for (size_t i = 0; i < daemon_resp.addresses.size(); ++i)
+      batching_records_cache.emplace_back(batching_detail{daemon_resp.addresses[i], daemon_resp.amounts[i]});
```

Comment:
```suggestion
      batching_records_cache.emplace_back(std::move(daemon_resp.addresses[i]), std::move(daemon_resp.amounts[i]));
```

I think that's better but brain is half-on atm.

Reply:
or alternatively use an iterator range based insert

Reply:
also see below regarding using a map rather than vector of struct.

Reply:
use std::map if you need determinism

Reply:
Random comment on this: `emplace` is intended to directly invoke the constructor, i.e. everything you pass to emplace ends up in a constructor.  So `whatever.emplace(T{x,y})` effectively ends up a `T{T{x,y}}` (which constructs a temporary, then passes it to another T's move construcctor), while a plain `emplace(x,y)` just does a single `T{x,y}` construction.